var/spack/repos/builtin/packages/lbann/package.py

```diff
@@ -357,181 +326,179 @@ def setup_build_environment(self, env):
+        return sys_type
 
-    @when("@:0.90,0.94:")
-    def cmake_args(self):
-        spec = self.spec
-        args = self.common_config_args
-        args.extend(
-            [
-                "-DCMAKE_CXX_STANDARD=17",
-                "-DLBANN_WITH_CNPY=%s" % ("+numpy" in spec),
-                "-DLBANN_DETERMINISTIC:BOOL=%s" % ("+deterministic" in spec),
-                "-DLBANN_WITH_HWLOC=%s" % ("+hwloc" in spec),
-                "-DLBANN_WITH_ALUMINUM:BOOL=%s" % ("+al" in spec),
-                "-DLBANN_WITH_ADDRESS_SANITIZER:BOOL=%s" % ("+asan" in spec),
-                "-DLBANN_WITH_BOOST:BOOL=%s" % ("+boost" in spec),
-                "-DLBANN_WITH_CONDUIT:BOOL=%s" % ("+conduit" in spec),
-                "-DLBANN_WITH_NVSHMEM:BOOL=%s" % ("+nvshmem" in spec),
-                "-DLBANN_WITH_FFT:BOOL=%s" % ("+fft" in spec),
-                "-DLBANN_WITH_ONEDNN:BOOL=%s" % ("+onednn" in spec),
-                "-DLBANN_WITH_ONNX:BOOL=%s" % ("+onnx" in spec),
-                "-DLBANN_WITH_EMBEDDED_PYTHON:BOOL=%s" % ("+python" in spec),
-                "-DLBANN_WITH_PYTHON_FRONTEND:BOOL=%s" % ("+pfe" in spec),
-                "-DLBANN_WITH_ROCTRACER:BOOL=%s" % ("+rocm +distconv" in spec),
-                "-DLBANN_WITH_TBINF=OFF",
-                "-DLBANN_WITH_UNIT_TESTING:BOOL=%s" % ("+unit_tests" in spec),
-                "-DLBANN_WITH_VISION:BOOL=%s" % ("+vision" in spec),
-                "-DLBANN_WITH_VTUNE:BOOL=%s" % ("+vtune" in spec),
-                "-DLBANN_DATATYPE={0}".format(spec.variants["dtype"].value),
-                "-DCEREAL_DIR={0}".format(spec["cereal"].prefix),
-                # protobuf is included by py-protobuf+cpp
-                "-DProtobuf_DIR={0}".format(spec["protobuf"].prefix),
-                "-Dprotobuf_MODULE_COMPATIBLE=ON",
-            ]
+    @property
+    def cache_name(self):
+        hostname = socket.gethostname()
+        if "SYS_TYPE" in env:
+            hostname = hostname.rstrip("1234567890")
+        return "LBANN_{0}_{1}-{2}-{3}@{4}.cmake".format(
+            hostname,
+            self.spec.version,
+            self._get_sys_type(self.spec),
+            self.spec.compiler.name,
+            self.spec.compiler.version,
         )
 
+    def initconfig_compiler_entries(self):
+        spec = self.spec
+        entries = super(Lbann, self).initconfig_compiler_entries()
+        entries.append(cmake_cache_string("CMAKE_CXX_STANDARD", "17"))
         if not spec.satisfies("^cmake@3.23.0"):
             # There is a bug with using Ninja generator in this version
             # of CMake
-            args.append("-DCMAKE_EXPORT_COMPILE_COMMANDS=ON")
+            entries.append(cmake_cache_option("CMAKE_EXPORT_COMPILE_COMMANDS", True))
+
+        entries.append(
+            cmake_cache_string("CMAKE_MAKE_PROGRAM", "{0}/ninja".format(spec["ninja"].prefix.bin))
+        )
+        entries.append(cmake_cache_string("CMAKE_GENERATOR", "Ninja"))
```

Comment:
Note that you already have `generator("ninja")` which should add this to the `cmake` command-line options by default (this should also occur for `CachedCMakePackage`s).

Reply:
@scheibelp I have pushed the definition of this into the cached_cmake.py package.